refactor: configure app mount point via `SCRIPT_NAME`

Midnighter opened the pull request:
I hope, I didn't forget anything but this should be all that's necessary to mount the app add a desired prefix.

kvikshaug commented on {{cookiecutter.project_slug}}/src/{{cookiecutter.project_module}}/settings.py:
```diff
@@ -46,8 +46,9 @@ def __init__(self):
         """
         self.DEBUG = True
         self.SECRET_KEY = os.urandom(24)
+        self.SCRIPT_NAME = os.environ.get('SCRIPT_NAME', '/')
```
`SCRIPT_NAME` is part of wsgi `environ`, it doesn't need to be set in the settings

Midnighter replied:
Did you test that? I thought I tried without and it wasn't working properly then but that day is kind of a blur to me.

kvikshaug replied:
Yes I tested it :)

kvikshaug commented on {{cookiecutter.project_slug}}/src/{{cookiecutter.project_module}}/settings.py:
```diff
@@ -105,4 +106,5 @@ def __init__(self):
         super().__init__()
         self.DEBUG = False
         self.SECRET_KEY = os.environ['SECRET_KEY']
+        self.SCRIPT_NAME = os.environ['SCRIPT_NAME']
```
`SCRIPT_NAME` is part of wsgi `environ`, it doesn't need to be set in the settings

kvikshaug commented:
Perhaps the root exception point is best remedied by simply changing the default development script_name to "".

kvikshaug commented:
> That said, SCRIPT_NAME is a first class configuration option for Flask so I don't feel like we're digging deep in the stack.

Actually, it comes from the [wsgi pep](https://www.python.org/dev/peps/pep-0333/#environ-variables), I can't find it documented as a config option in [the flask docs](http://flask.pocoo.org/docs/1.0/search/?q=SCRIPT_NAME&check_keywords=yes&area=default).

I suggest we move forward with this, updating the ingress accordingly and keeping it in `docker-compose.yml` available for override in `.env` but set to an empty string by default.

Midnighter commented:
Right, I mixed it up with the [gunicorn docs](http://docs.gunicorn.org/en/stable/faq.html?highlight=script_name#how-do-i-set-script-name). I believe I have addressed all your comments.

kvikshaug commented on {{cookiecutter.project_slug}}/docker-compose.yml:
```diff
@@ -17,6 +17,7 @@ services:
     environment:
       - ENVIRONMENT=${ENVIRONMENT:-development}
       - FLASK_APP=src/{{cookiecutter.project_module}}/wsgi.py
+      - SCRIPT_NAME=${SCRIPT_NAME:-}}}
```
Typo with closing brackets?
